src/mainActions.ts

```diff
@@ -1,144 +1,169 @@
-import * as electronOs from "os"
-import { ipcMain, dialog, BrowserWindow } from 'electron';
+import * as electronOs from "os";
+import { ipcMain, dialog, BrowserWindow } from "electron";
 
 import {
   ErrorInfo,
   OPEN_DIALOG_CHANNEL,
   OPEN_ERROR_DIALOG_CHANNEL,
   IMPORT_REPLY_CHANNEL,
-  SAVE_REPLY_CHANNEL, 
+  SAVE_REPLY_CHANNEL,
   JSON_IMPORT_REPLY_FORMAT,
   SQLITE_IMPORT_REPLY_FORMAT,
   SQLITE_SAVE_REPLY_FORMAT,
   SQLITE_PATH_FOR_JSON_REQUEST_FORMAT,
   UTIL_CHANNEL,
-} from './constants/IpcConnection';
+} from "./constants/IpcConnection";
 
-export function chooseSqlitePathForJsonImport(mainWindow: BrowserWindow, jsonFilePath: string){
+export function chooseSqlitePathForJsonImport(
+  mainWindow: BrowserWindow,
+  jsonFilePath: string
+) {
   console.log("Prompt to choose Sqlite path for JSON file");
 
   let defaultPath;
-  if (electronOs.platform() === 'linux') {
+  if (electronOs.platform() === "linux") {
     // remove the .json (or other) extention of the path.
-    defaultPath = jsonFilePath.substring(0, jsonFilePath.lastIndexOf('.')) + '.sqlite';
+    defaultPath =
+      jsonFilePath.substring(0, jsonFilePath.lastIndexOf(".")) + ".sqlite";
   } else {
     // FIXME: this is some ugly regex used to get filename with no extension.
     // see: https://stackoverflow.com/questions/4250364/how-to-trim-a-file-extension-from-a-string-in-javascript
-    defaultPath = jsonFilePath.replace(/^.*[\\/]/, '').replace(/\.[^/.]+$/, '');
+    defaultPath = jsonFilePath.replace(/^.*[\\/]/, "").replace(/\.[^/.]+$/, "");
   }
 
   // Immediately ask for a path to create & save the SQLite database
-  dialog.showSaveDialog(mainWindow, {
-    title: 'Save a SQLite Database File',
-    defaultPath: defaultPath,
-    filters: [{
-      name: 'SQLite File',
-      extensions: ['sqlite']
-    }]
-  }).then((sqliteFile) => {
-    const sqliteFilePath = sqliteFile.filePath;
-    if (sqliteFilePath === undefined) {
-      console.log("Sqlite file path isn't valid:", sqliteFilePath);
-      return;
-    }
-    const reply: JSON_IMPORT_REPLY_FORMAT = {
-      jsonFilePath,
-      sqliteFilePath
-    }
-    mainWindow.webContents.send(IMPORT_REPLY_CHANNEL.JSON, reply);
-  });
+  dialog
```

Comment:
We could have a default here that we can set, like a specific folder that we save the sqlite file in, to save time. This is a refinement of course, no need to do this in this PR.

Reply:
yeah makes sense, maybe settings kinda option for users to set the default path